src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -207,6 +190,84 @@ func (r *StatefulSetResource) rollingUpdate(
 	return nil
 }
 
+func (r *StatefulSetResource) rollingRestart(ctx context.Context, pod, artificialPod *corev1.Pod, newVolumes map[string]interface{}) error {
+	opts := []patch.CalculateOption{
+		patch.IgnoreStatusFields(),
+		ignoreKubernetesTokenVolumeMounts(),
+		ignoreDefaultToleration(),
+		ignoreExistingVolumes(newVolumes),
+	}
+
+	patchResult, err := patch.DefaultPatchMaker.Calculate(pod, artificialPod, opts...)
+	if err != nil {
+		return err
+	}
+
+	if patchResult.IsEmpty() {
+		return nil
+	}
+
+	var ordinal int64
+	ordinal, err = strconv.ParseInt(pod.Name[len(r.pandaCluster.Name)+1:], 10, 32)
```

Comment:
this is very hard to understand for me 😅 I would almost write a separate function for it and test it. So we're saying that the pod is named the same as pandacluster and then from the next character further down it's ordinal... 🤔 yeah I guess that's right...

Reply:
We have the same usage in https://github.com/redpanda-data/redpanda/blob/a6a021bc562f692eaf1793fcc8a6bb179318e374/src/go/k8s/controllers/redpanda/cluster_controller.go#L554

I can move the extraction of the ordinal to separated function.